### DISCUSSION TITLE
KP_ADDR 16, should it be changed if physical keypads?

### DISCUSSION THREAD

**Discussion by AlexandreUSA** [Dilbert66/esphome-vistaECP#27] [Q&A] [answered]
Hi, here is another question: Keypad address (KP_ADDR  16). Does it need to be unique in a install? I already have 3 physical keypads under the addresses 16, 17, and 18 in my Vista 21ip programmation. Do I need to bring the KP_ADDR to 19 or should I keep to 16? Thanks.

**Comment by Dilbert66** [comment, marked as answer]
Yes, it should be unique or the program will have problems sending commands to the panel.  You also need to enable the keypad address in the vista panel.   For address 19, it needs to be done in program section 192.  Look at your  vista 21 installer manual for details.  And you also need to set keypaddAddr: "19" in the yaml instead of keypadAddr "16".  You don't set kp_addr as the address in the yaml overrides it.

**Reply by AlexandreUSA** [reply]
Perfect. Worked like a charm. I changed the keypadAddr to 19 in the yaml file, updated my Vista 21ip configuration (*192, 1,0) and the system seems to be working fine.
